<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>R Notebook</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">10x-adipocyte-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/PytrikFolkertsma/10x-adipocyte-analysis">
    <span class="fa fa-github"></span>
     
    Source
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>


<p><strong>Last updated:</strong> 2019-01-01</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20181026)</code> </summary></p>
<p>The command <code>set.seed(20181026)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/tree/a7860ef56803686ca1401f9b7c94a815722a116e" target="_blank">a7860ef</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    output/10x-180504
    Ignored:    output/10x-180504-aligned
    Ignored:    output/10x-180504-aligned-metageneplot
    Ignored:    output/10x-180504-beforeQC
    Ignored:    output/10x-180504-beforeqc
    Ignored:    output/10x-180504-cca-discardedcells
    Ignored:    output/10x-180504-ccregout
    Ignored:    output/10x-180504-ccregout-aligned
    Ignored:    output/10x-180504-ccregout-cca-discardedcells
    Ignored:    output/10x-180831
    Ignored:    output/10x-180831-T1T2T3
    Ignored:    output/10x-180831-T4T5
    Ignored:    output/10x-180831-beforeqc
    Ignored:    output/10x-180831-notcleaned
    Ignored:    output/monocle/

Untracked files:
    Untracked:  tables/10x-180504-scmap-numbers

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/a7860ef56803686ca1401f9b7c94a815722a116e/analysis/10x-180504-scmap.Rmd" target="_blank">a7860ef</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2019-01-01
</td>
<td style="text-align:left;">
wflow_publish(c(“analysis/10x-180504-scmap.Rmd”))
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/456926a654fed74a351e84229ab96dae46742e21/analysis/10x-180504-scmap.Rmd" target="_blank">456926a</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2019-01-01
</td>
<td style="text-align:left;">
updated notebooks
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/1ee9efb40dd71758c60f6d5bb622fddfe1fd976e/analysis/10x-180504-scmap.Rmd" target="_blank">1ee9efb</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2019-01-01
</td>
<td style="text-align:left;">
updated notebooks
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Loading required package: cowplot</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre class="r"><code>library(scmap)</code></pre>
<pre><code>Creating a generic function for &#39;toJSON&#39; from package &#39;jsonlite&#39; in package &#39;googleVis&#39;</code></pre>
<pre class="r"><code>library(SingleCellExperiment)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:Matrix&#39;:

    colMeans, colSums, rowMeans, rowSums, which</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, cbind, colMeans,
    colnames, colSums, do.call, duplicated, eval, evalq, Filter,
    Find, get, grep, grepl, intersect, is.unsorted, lapply,
    lengths, Map, mapply, match, mget, order, paste, pmax,
    pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
    rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Matrix&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: DelayedArray</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>
Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    apply</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<p>Data loading and inspection of the metadata.</p>
<pre class="r"><code>load(&#39;/data/pub-others/tabula_muris/figshare/180126-facs/maca.seurat_obj.facs.figshare_180126.RData&#39;)
head(seurat_obj@meta.data)</code></pre>
<pre><code>                      nGene    nUMI    orig.ident                  cell
A12.D041914.3_8_M.1.1  4712 1221663 SeuratProject A12.D041914.3_8_M.1.1
B16.D041914.3_8_M.1.1  3652 1837897 SeuratProject B16.D041914.3_8_M.1.1
C18.D041914.3_8_M.1.1  6220 1703523 SeuratProject C18.D041914.3_8_M.1.1
D22.D041914.3_8_M.1.1  5322  949162 SeuratProject D22.D041914.3_8_M.1.1
F4.D041914.3_8_M.1.1   3036   59975 SeuratProject  F4.D041914.3_8_M.1.1
C19.D041914.3_8_M.1.1  5350  808935 SeuratProject C19.D041914.3_8_M.1.1
                       tissue cell_ontology_class cell_ontology_id
A12.D041914.3_8_M.1.1 Bladder    mesenchymal cell       CL:0008019
B16.D041914.3_8_M.1.1 Bladder        bladder cell       CL:1001319
C18.D041914.3_8_M.1.1 Bladder        bladder cell       CL:1001319
D22.D041914.3_8_M.1.1 Bladder        bladder cell       CL:1001319
F4.D041914.3_8_M.1.1  Bladder    mesenchymal cell       CL:0008019
C19.D041914.3_8_M.1.1 Bladder        bladder cell       CL:1001319
                              tissue_cell_type
A12.D041914.3_8_M.1.1 Bladder_mesenchymal cell
B16.D041914.3_8_M.1.1     Bladder_bladder cell
C18.D041914.3_8_M.1.1     Bladder_bladder cell
D22.D041914.3_8_M.1.1     Bladder_bladder cell
F4.D041914.3_8_M.1.1  Bladder_mesenchymal cell
C19.D041914.3_8_M.1.1     Bladder_bladder cell</code></pre>
<pre class="r"><code>sce_maca &lt;- as.SingleCellExperiment(seurat_obj)
all10x &lt;- readRDS(&#39;output/10x-180504&#39;)
sce_10x &lt;- as.SingleCellExperiment(all10x)

#convert maca gene names to uppercase to match 10x gene names
rowData(sce_maca)[&#39;feature_symbol&#39;] &lt;- unlist(lapply(rowData(sce_maca)$gene, function(x){return(toupper(x))}))
rowData(sce_10x)[&#39;feature_symbol&#39;] &lt;- rowData(sce_10x)$gene

counts(sce_10x) &lt;- as.matrix(counts(sce_10x))
logcounts(sce_10x) &lt;- as.matrix(logcounts(sce_10x))

counts(sce_maca) &lt;- as.matrix(counts(sce_maca))
logcounts(sce_maca) &lt;- as.matrix(logcounts(sce_maca))
sce_maca &lt;- selectFeatures(sce_maca, suppress_plot = FALSE)</code></pre>
<p><img src="figure/10x-180504-scmap.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Celltypes in the fat dataset</p>
<pre class="r"><code>seurat_obj@meta.data %&gt;% filter(tissue==&quot;Fat&quot;) %&gt;% distinct(tissue_cell_type)</code></pre>
<pre><code>                       tissue_cell_type
1                      Fat_myeloid cell
2                            Fat_T cell
3                            Fat_B cell
4                       Fat_granulocyte
5  Fat_mesenchymal stem cell of adipose
6                  Fat_endothelial cell
7               Fat_natural killer cell
8                   Fat_epithelial cell
9                        Fat_neutrophil
10               Fat_smooth muscle cell</code></pre>
<p>Subsetting and preparing the data.</p>
<pre class="r"><code>maca_fat &lt;- SubsetData(SetAllIdent(seurat_obj, id=&#39;tissue&#39;), ident.use=&quot;Fat&quot;)
sce_maca_fat &lt;- as.SingleCellExperiment(maca_fat)
rowData(sce_maca_fat)[&#39;feature_symbol&#39;] &lt;- unlist(lapply(rowData(sce_maca_fat)$gene, function(x){return(toupper(x))}))
counts(sce_maca_fat) &lt;- as.matrix(counts(sce_maca_fat))
logcounts(sce_maca_fat) &lt;- as.matrix(logcounts(sce_maca_fat))
sce_maca_fat &lt;- selectFeatures(sce_maca_fat, suppress_plot = FALSE)</code></pre>
<p><img src="figure/10x-180504-scmap.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Setting the right column for clustering.</p>
<pre class="r"><code>sce_maca_fat &lt;- indexCluster(sce_maca_fat, cluster_col = &#39;cell_ontology_class&#39;)</code></pre>
<p>Predicting cell types in our dataset.</p>
<pre class="r"><code>scmapCluster_results_fat &lt;- scmapCluster(
  projection = sce_10x, 
  index_list = list(
    sce_maca_fat = metadata(sce_maca_fat)$scmap_cluster_index
  ),
  threshold=0.5  #default=0.7 
)</code></pre>
<pre><code>Warning in setFeatures(projection, rownames(index)): Features
1190002H23RIK, 8430408G22RIK, ADH1, AW112010, C1RA, C4B, CAR4, CCL6,
CCL9, CCR2, CD2, CD24A, CD48, CD53, CXCR7, CYB5, CYBB, CYP4B1, CYP4F18,
ERCC-00009, ERCC-00108, F13A1, FCGR2B, FCGR3, GIMAP3, GIMAP6, GM11428,
GPIHBP1, H2-AA, H2-AB1, H2-D1, H2-DMA, H2-DMB1, H2-DMB2, H2-EB1, H2-K1, H2-
OB, H2-Q6, HMGCS2, IFI205, IFI27L2A, IL11RA1, LILRB4, LRRC33, LY6A, LY6C1,
LY86, LYZ2, MGL2, MMP23, MRC1, MS4A1, MS4A4B, MS4A4C, MS4A4D, MS4A6B,
MS4A6C, MT1, NEURL3, PECAM1, PGCP, RETNLA, SERPINB6A, SFPI1, SLFN2, SPNB2,
TPRGL, TRF are not present in the &#39;SCESet&#39; object and therefore were not
set.</code></pre>
<p>Number of predictions for each annotation for the whole dataset and for the mixture cluster.</p>
<pre class="r"><code>pred_fat &lt;- as.data.frame(table(scmapCluster_results_fat$scmap_cluster_labs))
pred_fat &lt;- pred_fat[order(-pred_fat$Freq),]

pred_mixt_fat &lt;- as.data.frame(table(scmapCluster_results_fat$scmap_cluster_labs[which(colData(sce_10x)$res.0.5 %in% 12), &#39;sce_maca_fat&#39;]))

scmap_nr_predictions &lt;- merge(pred_fat, pred_mixt_fat, by=&#39;Var1&#39;, suffixes=c(&#39;.total&#39;, &#39;.mixture&#39;))
scmap_nr_predictions</code></pre>
<pre><code>                              Var1 Freq.total Freq.mixture
1                  epithelial cell        861          723
2 mesenchymal stem cell of adipose      43909           78
3               smooth muscle cell        749           18
4                       unassigned      10852          320</code></pre>
<p>Interestingly, a lot of epithelial cell predictions in the mixture cluster and not that much mesenchymal stem cell predictions.</p>
<pre class="r"><code>predicted_labels_fat &lt;- as.data.frame(
    row.names=rownames(sce_10x@colData), 
    x=as.vector(scmapCluster_results_fat$scmap_cluster_labs))
names(predicted_labels_fat) &lt;- &#39;predicted_labels_fat&#39;
all10x &lt;- AddMetaData(all10x, metadata=predicted_labels_fat, col.name=&#39;predicted_labels_fat&#39;)
t1 &lt;- TSNEPlot(all10x, group.by=&#39;predicted_labels_fat&#39;, pt.size=0.1)</code></pre>
<p><img src="figure/10x-180504-scmap.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(&quot;/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/plots_slides/scmap.pdf&quot;, t1, base_width=8, base_height = 5)</code></pre>
<pre class="r"><code>TSNEPlot(all10x, group.by=&#39;sample_name&#39;, pt.size=0.1, do.label=T)</code></pre>
<p><img src="figure/10x-180504-scmap.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#saveRDS(all10x, &#39;output/10x-180504&#39;)</code></pre>
<pre class="r"><code>write.table(scmap_nr_predictions, &#39;tables/10x-180504-scmap-numbers&#39;)</code></pre>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Storage

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bindrcpp_0.2.2             dplyr_0.7.6               
 [3] SingleCellExperiment_1.0.0 SummarizedExperiment_1.8.1
 [5] DelayedArray_0.4.1         matrixStats_0.54.0        
 [7] Biobase_2.38.0             GenomicRanges_1.30.3      
 [9] GenomeInfoDb_1.14.0        IRanges_2.12.0            
[11] S4Vectors_0.16.0           BiocGenerics_0.24.0       
[13] scmap_1.1.5                Seurat_2.3.4              
[15] Matrix_1.2-14              cowplot_0.9.3             
[17] ggplot2_3.0.0             

loaded via a namespace (and not attached):
  [1] Rtsne_0.13             colorspace_1.3-2       class_7.3-14          
  [4] modeltools_0.2-22      ggridges_0.5.0         mclust_5.4.1          
  [7] rprojroot_1.3-2        htmlTable_1.12         XVector_0.18.0        
 [10] base64enc_0.1-3        rstudioapi_0.7         proxy_0.4-22          
 [13] flexmix_2.3-14         bit64_0.9-7            mvtnorm_1.0-8         
 [16] codetools_0.2-15       splines_3.4.3          R.methodsS3_1.7.1     
 [19] robustbase_0.93-2      knitr_1.20             Formula_1.2-3         
 [22] jsonlite_1.5           workflowr_1.1.1        ica_1.0-2             
 [25] cluster_2.0.7-1        kernlab_0.9-27         png_0.1-7             
 [28] R.oo_1.22.0            compiler_3.4.3         httr_1.3.1            
 [31] googleVis_0.6.2        backports_1.1.2        assertthat_0.2.0      
 [34] lazyeval_0.2.1         lars_1.2               acepack_1.4.1         
 [37] htmltools_0.3.6        tools_3.4.3            igraph_1.2.2          
 [40] GenomeInfoDbData_1.0.0 gtable_0.2.0           glue_1.3.0            
 [43] RANN_2.6               reshape2_1.4.3         Rcpp_0.12.18          
 [46] trimcluster_0.1-2.1    gdata_2.18.0           ape_5.1               
 [49] nlme_3.1-137           iterators_1.0.10       fpc_2.1-11.1          
 [52] gbRd_0.4-11            lmtest_0.9-36          stringr_1.3.1         
 [55] irlba_2.3.2            gtools_3.8.1           DEoptimR_1.0-8        
 [58] zlibbioc_1.24.0        MASS_7.3-50            zoo_1.8-3             
 [61] scales_1.0.0           doSNOW_1.0.16          RColorBrewer_1.1-2    
 [64] yaml_2.2.0             reticulate_1.10        pbapply_1.3-4         
 [67] gridExtra_2.3          rpart_4.1-13           segmented_0.5-3.0     
 [70] latticeExtra_0.6-28    stringi_1.2.4          randomForest_4.6-14   
 [73] foreach_1.4.4          e1071_1.7-0            checkmate_1.8.5       
 [76] caTools_1.17.1.1       bibtex_0.4.2           Rdpack_0.9-0          
 [79] SDMTools_1.1-221       rlang_0.2.2            pkgconfig_2.0.2       
 [82] dtw_1.20-1             prabclus_2.2-6         bitops_1.0-6          
 [85] evaluate_0.11          lattice_0.20-35        ROCR_1.0-7            
 [88] purrr_0.2.5            bindr_0.1.1            labeling_0.3          
 [91] htmlwidgets_1.2        bit_1.1-14             tidyselect_0.2.4      
 [94] plyr_1.8.4             magrittr_1.5           R6_2.2.2              
 [97] snow_0.4-2             gplots_3.0.1           Hmisc_4.1-1           
[100] pillar_1.3.0           whisker_0.3-2          foreign_0.8-71        
[103] withr_2.1.2            fitdistrplus_1.0-9     mixtools_1.1.0        
[106] RCurl_1.95-4.11        survival_2.42-6        nnet_7.3-12           
[109] tsne_0.1-3             tibble_1.4.2           crayon_1.3.4          
[112] hdf5r_1.0.0            KernSmooth_2.23-15     rmarkdown_1.10        
[115] grid_3.4.3             data.table_1.11.4      git2r_0.23.0          
[118] metap_1.0              digest_0.6.16          diptest_0.75-7        
[121] tidyr_0.8.1            R.utils_2.7.0          munsell_0.5.0         </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
