<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>R Notebook</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">10x-adipocyte-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/PytrikFolkertsma/10x-adipocyte-analysis">
    <span class="fa fa-github"></span>
     
    Source
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-05-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>10x-adipocyte-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181026code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181026)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181026code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181026)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompytrikfolkertsma10xadipocyteanalysistree1de6405764932f25f5ba26b49f9aceb0b6526eb6targetblank1de6405a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/tree/1de6405764932f25f5ba26b49f9aceb0b6526eb6" target="_blank">1de6405</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcompytrikfolkertsma10xadipocyteanalysistree1de6405764932f25f5ba26b49f9aceb0b6526eb6targetblank1de6405a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    figures/
    Ignored:    output/bulk_analysis/
    Ignored:    output/demuxlet/
    Ignored:    output/markergenes/
    Ignored:    output/monocle/
    Ignored:    output/seurat_objects/
    Ignored:    output/velocyto/
    Ignored:    output/wgcna/
    Ignored:    tables/

Untracked files:
    Untracked:  .rstudio_old10/
    Untracked:  10x-adipocyte-analysis-copy.Rproj
    Untracked:  analysis/.ipynb_checkpoints/velocyto_notebook_180831-checkpoint.ipynb
    Untracked:  analysis/10-180831-monocle-per-depot.Rmd
    Untracked:  analysis/10x-180831-BEAM-heatmap.Rmd
    Untracked:  analysis/10x-180831-pseudotime.Rmd
    Untracked:  code/BEAM-heatmaps.R
    Untracked:  code/BEAM_gsea.R
    Untracked:  code/colors.R

Unstaged changes:
    Deleted:    10x-adipocyte-analysis.Rproj
    Modified:   analysis/10x-180504-DEGs-depots.Rmd
    Modified:   analysis/10x-180504-alignment.Rmd
    Modified:   analysis/10x-180504-depot-markers.Rmd
    Modified:   analysis/10x-180831-BATLAS.Rmd
    Modified:   analysis/10x-180831-TF_analysis.Rmd
    Modified:   analysis/10x-180831-beamGOplot.Rmd
    Modified:   analysis/10x-180831-colors.Rmd
    Modified:   analysis/10x-180831-figures.Rmd
    Modified:   analysis/10x-180831-general-analysis.Rmd
    Modified:   analysis/10x-180831-supplementary_figures.Rmd
    Modified:   analysis/velocyto_notebook_180504.ipynb
    Modified:   analysis/velocyto_notebook_180831.ipynb
    Deleted:    code/REMOVE/find-brown-sample-markers-180504-REMOVE.R
    Deleted:    code/REMOVE/find-white-sample-markers-180504-REMOVE.R
    Deleted:    code/REMOVE/get-genes-monocle-180831-REMOVE.R
    Modified:   code/compute-genelists-monocle-depots.R
    Modified:   code/find-depot-markers-180504.R
    Modified:   code/find-markers.R
    Modified:   code/preprocess-data.R
    Modified:   code/run-monocle.R
    Modified:   code/velocyto_preprocess.py

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pytrikfolkertsma/10x-adipocyte-analysis/158bede2ed6eebe5cfc3a64d26645f5bd931d113/docs/10x-180831-demuxlet.html" target="_blank">158bede</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/7b6d1975706285933a3e2e241ac6bff0f8cef3d8/analysis/10x-180831-demuxlet.Rmd" target="_blank">7b6d197</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
<td>
added nCells tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pytrikfolkertsma/10x-adipocyte-analysis/ac858a14c8433b580c61f68ed87693ec4b61924a/docs/10x-180831-demuxlet.html" target="_blank">ac858a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/c22f3f2b6d0594214bfda38164f039e0b2965e03/analysis/10x-180831-demuxlet.Rmd" target="_blank">c22f3f2</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
<td>
demuxlet results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pytrikfolkertsma/10x-adipocyte-analysis/ea82b7d5c47d00654ca9002915b98e650a445968/docs/10x-180831-demuxlet.html" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/a8a36a6b2116a6da6c12bfa9d8739a4a2530e6c7/analysis/10x-180831-demuxlet.Rmd" target="_blank">a8a36a6</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
<td>
demuxlet results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pytrikfolkertsma/10x-adipocyte-analysis/07bf255afb540f8441a5028ff0ca3ac5f289e3bb/docs/10x-180831-demuxlet.html" target="_blank">07bf255</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ca15d0e01df51bee7f48941a38eb6bbceaf5518f/analysis/10x-180831-demuxlet.Rmd" target="_blank">ca15d0e</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
<td>
demuxlet notebook
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/03050a1a452f80fdd42d3f94fa63477fc0f927ac/analysis/10x-180831-demuxlet.Rmd" target="_blank">03050a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
<td>
analysis updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(Seurat)
library(dplyr)
library(tidyr)
library(ggplot2)</code></pre>
<pre class="r"><code>data &lt;- readRDS(&#39;output/seurat_objects/180831/10x-180831-notcleaned&#39;)</code></pre>
<pre class="r"><code>getDemuxletForSample &lt;- function(i, outputdir){
  demuxlet &lt;- read.table(paste(&#39;output/demuxlet/190110_demuxlet_new_genotypes/demuxlet_out/&#39;, outputdir, &#39;/180831_10x_s&#39;, i, &#39;.best&#39;, sep=&#39;&#39;), header=T)

  demuxlet$correct_barcode &lt;- paste(unlist(sapply(strsplit(as.character(demuxlet$BARCODE), &#39;-&#39;), &#39;[[&#39;, 1)), &#39;-&#39;, i, sep=&#39;&#39;)
  rownames(demuxlet) &lt;- demuxlet$correct_barcode
  cells &lt;- rownames(data@meta.data)[data@meta.data$timepoint == paste(&#39;T&#39;, i, sep=&#39;&#39;)]
  demuxlet_filtered &lt;- demuxlet[demuxlet$correct_barcode %in% cells, ]
  demuxlet_filtered[&#39;sample&#39;] &lt;- i
  
  sng_dbl_abm &lt;- sapply(strsplit(as.character(demuxlet_filtered$BEST), &#39;-&#39;), &#39;[[&#39;, 1)
  demuxlet_filtered[&#39;sng_dbl_amb&#39;] &lt;- sng_dbl_abm
  
  #add counts singlets, doublets and ambiguous
  demuxlet_filtered[c(&#39;SNG&#39;, &#39;DBL&#39;, &#39;AMB&#39;)] &lt;- 0
  demuxlet_filtered &lt;- demuxlet_filtered %&gt;% mutate(value=1) %&gt;% spread(sng_dbl_amb, value, fill=0)
  demuxlet_filtered[&#39;sng_dbl_amb&#39;] &lt;- sng_dbl_abm
  
  return(demuxlet_filtered)
  
}

getAllDemuxletResults &lt;- function(outdir){
  demuxlet_list &lt;- list()
  for (i in 1:5){
    demuxlet &lt;- getDemuxletForSample(i, outdir)
    demuxlet_list[[i]] &lt;- demuxlet
  }
  demuxlet_all &lt;- do.call(rbind, unname(demuxlet_list))
  demuxlet_all$label &lt;- as.character(demuxlet_all$BEST)
  demuxlet_all$label[startsWith(demuxlet_all$label, &#39;DBL&#39;)] &lt;- &#39;DBL&#39;
  demuxlet_all$label[startsWith(demuxlet_all$label, &#39;AMB&#39;)] &lt;- &#39;AMB&#39;

  df_sda &lt;- as.data.frame(aggregate(demuxlet_all[c(&#39;SNG&#39;, &#39;DBL&#39;, &#39;AMB&#39;)], by=list(sample=demuxlet_all$sample), FUN=sum))
  df_snp &lt;- as.data.frame(aggregate(demuxlet_all[c(&#39;N.SNP&#39;)], by=list(sample=demuxlet_all$sample), FUN=mean))
  
  print(cbind(df_sda, df_snp$N.SNP))
  
  print(&#39;Total number of SNG, DBL and AMB:&#39;)
  print(table(demuxlet_all$sng_dbl_amb))
  
  print(paste(&#39;Total average N.SNP:&#39;, mean(demuxlet_all$N.SNP)))
  hist(demuxlet_all$N.SNP)
  
  return(demuxlet_all)
}</code></pre>
<div id="demuxlet-with-qced-vcf-file" class="section level1">
<h1>Demuxlet with QCed VCF file</h1>
<pre class="r"><code>demuxlet_all_qc &lt;- getAllDemuxletResults(&#39;demuxlet_plink_bed-updated&#39;)</code></pre>
<pre><code>  sample  SNG DBL AMB df_snp$N.SNP
1      1 3074 359   0     89.41159
2      2 5220 311   0     78.10541
3      3 5761 289   0     90.71223
4      4 4058 432   2     51.57124
5      5 6659 675  31     40.57733
[1] &quot;Total number of SNG, DBL and AMB:&quot;

  AMB   DBL   SNG 
   33  2066 24772 
[1] &quot;Total average N.SNP: 67.6666294518254&quot;</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-4-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-4-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-2.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-2.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-3.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-3.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-4">
Past versions of unnamed-chunk-5-4.png
</button>
</p>
<div id="fig-unnamed-chunk-5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-4.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-4.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-5">
Past versions of unnamed-chunk-5-5.png
</button>
</p>
<div id="fig-unnamed-chunk-5-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-5.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-5.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-6">
Past versions of unnamed-chunk-5-6.png
</button>
</p>
<div id="fig-unnamed-chunk-5-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-6.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-5-6.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/fig1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig1-1">
Past versions of fig1-1.png
</button>
</p>
<div id="fig-fig1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/fig1-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/fig1-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="demuxlet-with-qced-vcf-exons-only" class="section level1">
<h1>Demuxlet with QCed VCF, exons only</h1>
<pre class="r"><code>demuxlet_all_qc_exons &lt;- getAllDemuxletResults(&#39;demuxlet_plink_bed-updated.exon_only.recode&#39;)</code></pre>
<pre><code>  sample  SNG DBL AMB df_snp$N.SNP
1      1 3056 377   0     68.64433
2      2 5153 376   2     56.22528
3      3 5721 329   0     66.24860
4      4 3957 518  17     35.26870
5      5 6460 749 156     26.79063
[1] &quot;Total number of SNG, DBL and AMB:&quot;

  AMB   DBL   SNG 
  175  2349 24347 
[1] &quot;Total average N.SNP: 48.4977112872614&quot;</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-6-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-6-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-2.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-2.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-3">
Past versions of unnamed-chunk-7-3.png
</button>
</p>
<div id="fig-unnamed-chunk-7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-3.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-3.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-4">
Past versions of unnamed-chunk-7-4.png
</button>
</p>
<div id="fig-unnamed-chunk-7-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-4.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-4.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-5">
Past versions of unnamed-chunk-7-5.png
</button>
</p>
<div id="fig-unnamed-chunk-7-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-5.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-5.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-6">
Past versions of unnamed-chunk-7-6.png
</button>
</p>
<div id="fig-unnamed-chunk-7-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-6.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-7-6.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/fig2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2-1">
Past versions of fig2-1.png
</button>
</p>
<div id="fig-fig2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/fig2-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/fig2-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="demuxlet-with-qced-vcf-imputed-snps" class="section level1">
<h1>Demuxlet with QCed VCF + imputed SNPs</h1>
<pre class="r"><code>demuxlet_all_qc_imputed &lt;- getAllDemuxletResults(&#39;demuxlet_chr1_22_combined.qc_r2_maf.recode&#39;)</code></pre>
<pre><code>  sample  SNG  DBL AMB df_snp$N.SNP
1      1 2948  485   0    146.00524
2      2 5065  465   1    147.65811
3      3 5593  457   0    165.16992
4      4 4015  477   0    379.63268
5      5 5846 1519   0     86.84942
[1] &quot;Total number of SNG, DBL and AMB:&quot;

  AMB   DBL   SNG 
    1  3403 23467 
[1] &quot;Total average N.SNP: 173.501804919802&quot;</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-8-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-8-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-2.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-2.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-3.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-3.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-4">
Past versions of unnamed-chunk-9-4.png
</button>
</p>
<div id="fig-unnamed-chunk-9-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-4.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-4.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-5">
Past versions of unnamed-chunk-9-5.png
</button>
</p>
<div id="fig-unnamed-chunk-9-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-5.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-5.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-6">
Past versions of unnamed-chunk-9-6.png
</button>
</p>
<div id="fig-unnamed-chunk-9-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-6.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-9-6.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/fig3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3-1">
Past versions of fig3-1.png
</button>
</p>
<div id="fig-fig3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/fig3-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/fig3-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="demuxlet-with-qced-vcf-imputed-snps-exons-only" class="section level1">
<h1>Demuxlet with QCed VCF + imputed SNPs, exons only</h1>
<pre class="r"><code>demuxlet_all_qc_exons_imputed &lt;- getAllDemuxletResults(&#39;demuxlet_chr1_22_combined.qc_r2_maf_exon.recode&#39;)</code></pre>
<pre><code>  sample  SNG  DBL AMB df_snp$N.SNP
1      1 2890  543   0     75.05884
2      2 5211  320   0    223.39487
3      3 5474  576   0     76.20165
4      4 3714  778   0    154.23753
5      5 6139 1226   0    104.61779
[1] &quot;Total number of SNG, DBL and AMB:&quot;

  DBL   SNG 
 3443 23428 
[1] &quot;Total average N.SNP: 127.186892932902&quot;</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-10-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-10-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-2.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-2.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-3.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-3.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-4">
Past versions of unnamed-chunk-11-4.png
</button>
</p>
<div id="fig-unnamed-chunk-11-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-4.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-11-4.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/fig4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4-1">
Past versions of fig4-1.png
</button>
</p>
<div id="fig-fig4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/fig4-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/fig4-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>demuxlet_list &lt;- list()
for (i in 1:5){
   demuxlet &lt;- getDemuxletForSample(i, &#39;demuxlet_chr1_22_combined.qc_r2_maf_exon.recode&#39;)
   demuxlet_list[[i]] &lt;- demuxlet
}
demuxlet_all &lt;- do.call(rbind, unname(demuxlet_list))

demuxlet_all$label &lt;- as.character(demuxlet_all$BEST)
demuxlet_all$label[startsWith(demuxlet_all$label, &#39;DBL&#39;)] &lt;- &#39;DBL&#39;
demuxlet_all$label[startsWith(demuxlet_all$label, &#39;AMB&#39;)] &lt;- &#39;AMB&#39;
rownames(demuxlet_all) &lt;- demuxlet_all$correct_barcode
 
data &lt;- AddMetaData(data, as.vector(demuxlet_all[&#39;label&#39;]))
TSNEPlot(data, group.by=&#39;label&#39;, pt.size=0.1)</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ac858a14c8433b580c61f68ed87693ec4b61924a/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-12-1.png" target="_blank">ac858a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-12-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-12-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>data@meta.data$label[is.na(data@meta.data$label)] &lt;- &quot;AMB&quot;

depot &lt;- unlist(lapply(data@meta.data$label, function(x){
  if (x == &#39;SNG-13a_13a&#39;){
    return(&#39;Subq&#39;)
  } else if (x == &#39;SNG-1AF_1AF&#39;){
    return(&#39;Peri&#39;)
  } else if (x == &#39;SNG-44B_44B&#39;){
    return(&#39;Visce&#39;)
  } else if (x == &#39;SNG-BAT14_BAT14&#39;){
    return(&#39;Supra&#39;)
  } else {
    return(&#39;DBL&#39;)
  }
}))

data@meta.data[&#39;depot&#39;] &lt;- depot
DimPlot(data, reduction=&#39;tsne&#39;, group.by=&#39;depot&#39;, pt.size=0.1)</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ac858a14c8433b580c61f68ed87693ec4b61924a/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-13-1.png" target="_blank">ac858a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-13-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Filter out doublets:</p>
<pre class="r"><code>data_cleaned &lt;- SubsetData(data, cells.use=rownames(data@meta.data)[data@meta.data$depot != &#39;DBL&#39;])</code></pre>
<pre class="r"><code>DimPlot(data_cleaned, reduction=&#39;tsne&#39;, group.by=&#39;depot&#39;, pt.size=0.1)</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ac858a14c8433b580c61f68ed87693ec4b61924a/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-15-1.png" target="_blank">ac858a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-15-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/bfc0e415767c4cbe19e428a1a27b9647a5f6d88d/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-15-1.png" target="_blank">bfc0e41</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>type &lt;- unlist(lapply(as.vector(data_cleaned@meta.data$depot), function(x){
  if (x == &#39;Subq&#39; || x == &#39;Visce&#39;){
    return(&#39;white&#39;)
  } else {
    return(&#39;brown&#39;)
  }
}))
data_cleaned@meta.data[&#39;type&#39;] &lt;- type
DimPlot(data_cleaned, reduction=&#39;tsne&#39;, group.by=&#39;type&#39;, pt.size=0.1)</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ac858a14c8433b580c61f68ed87693ec4b61924a/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-16-1.png" target="_blank">ac858a1</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/ea82b7d5c47d00654ca9002915b98e650a445968/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-16-1.png" target="_blank">ea82b7d</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>paste(&#39;Nr of cells before removing doublet and ambiguous:&#39;, length(rownames(data@meta.data)))</code></pre>
<pre><code>[1] &quot;Nr of cells before removing doublet and ambiguous: 26872&quot;</code></pre>
<pre class="r"><code>paste(&#39;Nr of cells in cleaned up data:&#39;, length(rownames(data_cleaned@meta.data)))</code></pre>
<pre><code>[1] &quot;Nr of cells in cleaned up data: 23428&quot;</code></pre>
<p>Number of cells per donor and timepoint.</p>
<pre class="r"><code>n_cells &lt;- as.data.frame(as.matrix(ftable(data_cleaned@meta.data$timepoint, data_cleaned@meta.data$depot)))

n_cells[nrow(n_cells), ] &lt;- colSums(n_cells)
rownames(n_cells)[length(rownames(n_cells))] &lt;- &#39;total&#39;

n_cells[,ncol(n_cells)+1] &lt;- rowSums(n_cells)
names(n_cells)[length(names(n_cells))] &lt;- &#39;total&#39;

n_cells</code></pre>
<pre><code>      Peri Subq Supra Visce total
T1     692  750   608   840  2890
T2    1273 1336  1351  1251  5211
T3    1259 1605  1118  1492  5474
T4    1236  762   857   859  3714
T5    1139 1816  1640  1544  6139
total 5599 6269  5574  5986 23428</code></pre>
<pre class="r"><code>#write.table(n_cells, &#39;../tables/tables_paper/supplementary_tables/10x-180831-ncells.tsv&#39;, sep=&#39;\t&#39;, quote=F, col.names=NA)</code></pre>
<pre class="r"><code>df &lt;- merge(demuxlet_all_qc_exons_imputed, data@meta.data, by.x=&#39;correct_barcode&#39;, by.y=&#39;row.names&#39;)

n_cells_timepoint_dbl_sng &lt;- as.data.frame(as.matrix(ftable(df$timepoint, df$sng_dbl_amb)))
ncells_timepoint_snps &lt;- aggregate(df$N.SNP, by=list(timepoint=df$timepoint), FUN=mean)
n_cells_combined &lt;- merge(n_cells_timepoint_dbl_sng, ncells_timepoint_snps, by.x=&#39;row.names&#39;, by.y=&#39;timepoint&#39;)
rownames(n_cells_combined) &lt;- n_cells_combined$Row.names
n_cells_combined[&#39;Row.names&#39;] &lt;- NULL
names(n_cells_combined)[length(names(n_cells_combined))] &lt;- &#39;avg N.SNP&#39;

#n_cells_combined[nrow(n_cells_combined)+1, ] &lt;- colSums(n_cells_combined)
#rownames(n_cells_combined)[length(rownames(n_cells_combined))] &lt;- &#39;total&#39;

#n_cells_combined[,ncol(n_cells_combined)+1] &lt;- rowSums(n_cells_combined)
#names(n_cells_combined)[length(names(n_cells_combined))] &lt;- &#39;total&#39;</code></pre>
<pre class="r"><code>#write.table(n_cells_combined, &#39;../tables/tables_paper/supplementary_tables/10x-180831-ncells_demuxlet.tsv&#39;, sep=&#39;\t&#39;, quote=F, col.names=NA)</code></pre>
<pre class="r"><code>#data@meta.data[&#39;depot&#39;] &lt;- substr(data@meta.data$sample_name, 1, nchar(data@meta.data$sample_name)-2)
#saveRDS(data, &#39;../../10x-adipocyte-analysis/output/10x-180831&#39;)</code></pre>
</div>
<div id="summary-of-results" class="section level1">
<h1>Summary of results</h1>
<pre class="r"><code>demuxlet_all_qc[&#39;vcf&#39;] &lt;- &#39;qc&#39;
demuxlet_all_qc_exons[&#39;vcf&#39;] &lt;- &#39;qc_exons&#39;
demuxlet_all_qc_imputed[&#39;vcf&#39;] &lt;- &#39;qc_imputed&#39;
demuxlet_all_qc_exons_imputed[&#39;vcf&#39;] &lt;- &#39;qc_exons_imputed&#39;
demuxlet_all &lt;- rbind(demuxlet_all_qc, demuxlet_all_qc_exons, demuxlet_all_qc_exons_imputed, demuxlet_all_qc_imputed)

test &lt;- aggregate(demuxlet_all[&#39;sng_dbl_amb&#39;], by=list(demuxlet_all$vcf, demuxlet_all$sng_dbl_amb), FUN=length)

df_aggregated &lt;- aggregate(demuxlet_all[c(&#39;SNG&#39;, &#39;DBL&#39;, &#39;AMB&#39;)], by=list(VCF=demuxlet_all$vcf, timepoint=demuxlet_all$sample), FUN=sum)

df_sng_dbl_amb_vcf &lt;- aggregate(demuxlet_all[c(&#39;SNG&#39;, &#39;DBL&#39;, &#39;AMB&#39;)], by=list(VCF=demuxlet_all$vcf), FUN=sum)</code></pre>
<p>Number of SNPs, SNG, DBL, AMB</p>
<pre class="r"><code>df_snps_vcf &lt;- aggregate(demuxlet_all[&#39;N.SNP&#39;], by=list(VCF=demuxlet_all$vcf), FUN=mean)
df_sng_dbl_amb_vcf[&#39;N.SNP&#39;] &lt;- df_snps_vcf$N.SNP
as.data.frame(df_sng_dbl_amb_vcf)</code></pre>
<pre><code>               VCF   SNG  DBL AMB     N.SNP
1               qc 24772 2066  33  67.66663
2         qc_exons 24347 2349 175  48.49771
3 qc_exons_imputed 23428 3443   0 127.18689
4       qc_imputed 23467 3403   1 173.50180</code></pre>
<pre class="r"><code>df_snps_vcf_timepoint &lt;- aggregate(demuxlet_all[&#39;N.SNP&#39;], by=list(VCF=demuxlet_all$vcf, timepoint=demuxlet_all$sample), FUN=mean)

ggplot(df_snps_vcf_timepoint, aes(x=VCF, y=N.SNP, fill=factor(timepoint))) + 
  geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
  labs(title=&#39;Number of SNPs&#39;, fill=&#39;timepoint&#39;, x=&#39;VCF file&#39;, y=&#39;&#39;) +
  theme_gray() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) </code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/158bede2ed6eebe5cfc3a64d26645f5bd931d113/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-25-1.png" target="_blank">158bede</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Number of singlets, doublets and ambiguous per VCF file.</p>
<pre class="r"><code>ggplot(df_aggregated, aes(x=VCF, y=SNG, fill=factor(timepoint))) + 
  geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
  labs(title=&#39;Number of singlets&#39;, fill=&#39;timepoint&#39;, x=&#39;VCF file&#39;, y=&#39;&#39;) +
  theme_gray() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/158bede2ed6eebe5cfc3a64d26645f5bd931d113/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-26-1.png" target="_blank">158bede</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df_aggregated, aes(x=VCF, y=DBL, fill=factor(timepoint))) + 
  geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
  labs(title=&#39;Number of doublets&#39;, fill=&#39;timepoint&#39;, x=&#39;VCF file&#39;, y=&#39;&#39;) +
  theme_gray() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/158bede2ed6eebe5cfc3a64d26645f5bd931d113/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-27-1.png" target="_blank">158bede</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df_aggregated, aes(x=VCF, y=AMB, fill=factor(timepoint))) + 
  geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
  labs(title=&#39;Number of ambiguous&#39;, fill=&#39;timepoint&#39;, x=&#39;VCF file&#39;, y=&#39;&#39;) +
  theme_gray() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/10x-180831-demuxlet.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pytrikfolkertsma/10x-adipocyte-analysis/blob/158bede2ed6eebe5cfc3a64d26645f5bd931d113/docs/figure/10x-180831-demuxlet.Rmd/unnamed-chunk-28-1.png" target="_blank">158bede</a>
</td>
<td>
Pytrik Folkertsma
</td>
<td>
2019-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Percentage doublets</p>
<pre class="r"><code>paste(&#39;QCed:&#39;,sum(demuxlet_all_qc$DBL) / length(demuxlet_all_qc$DBL))</code></pre>
<pre><code>[1] &quot;QCed: 0.0768858620817982&quot;</code></pre>
<pre class="r"><code>paste(&#39;QCed, exons only:&#39;, sum(demuxlet_all_qc_exons$DBL) / length(demuxlet_all_qc_exons$DBL))</code></pre>
<pre><code>[1] &quot;QCed, exons only: 0.0874176621636709&quot;</code></pre>
<pre class="r"><code>paste(&#39;QCed + imputed:&#39;, sum(demuxlet_all_qc_imputed$DBL) / length(demuxlet_all_qc_imputed$DBL))</code></pre>
<pre><code>[1] &quot;QCed + imputed: 0.126642104871423&quot;</code></pre>
<pre class="r"><code>paste(&#39;QCed + imputed, exons only&#39;, sum(demuxlet_all_qc_exons_imputed$DBL) / length(demuxlet_all_qc_exons_imputed$DBL))</code></pre>
<pre><code>[1] &quot;QCed + imputed, exons only 0.128130698522571&quot;</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Storage

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_0.8.3   dplyr_0.8.0.1 Seurat_2.3.4  Matrix_1.2-17 cowplot_0.9.4
[6] ggplot2_3.1.0

loaded via a namespace (and not attached):
  [1] Rtsne_0.15          colorspace_1.4-1    class_7.3-15       
  [4] modeltools_0.2-22   ggridges_0.5.1      mclust_5.4.3       
  [7] rprojroot_1.3-2     htmlTable_1.13.1    base64enc_0.1-3    
 [10] fs_1.2.7            rstudioapi_0.10     proxy_0.4-23       
 [13] npsurv_0.4-0        flexmix_2.3-15      bit64_0.9-7        
 [16] mvtnorm_1.0-10      codetools_0.2-16    splines_3.5.3      
 [19] R.methodsS3_1.7.1   lsei_1.2-0          robustbase_0.93-4  
 [22] knitr_1.22          jsonlite_1.6        Formula_1.2-3      
 [25] workflowr_1.2.0     ica_1.0-2           cluster_2.0.7-1    
 [28] kernlab_0.9-27      png_0.1-7           R.oo_1.22.0        
 [31] compiler_3.5.3      httr_1.4.0          backports_1.1.3    
 [34] assertthat_0.2.1    lazyeval_0.2.2      lars_1.2           
 [37] acepack_1.4.1       htmltools_0.3.6     tools_3.5.3        
 [40] igraph_1.2.4        gtable_0.3.0        glue_1.3.1         
 [43] reshape2_1.4.3      RANN_2.6.1          Rcpp_1.0.1         
 [46] trimcluster_0.1-2.1 gdata_2.18.0        ape_5.3            
 [49] nlme_3.1-137        iterators_1.0.10    fpc_2.1-11.1       
 [52] gbRd_0.4-11         lmtest_0.9-36       xfun_0.5           
 [55] stringr_1.4.0       irlba_2.3.3         gtools_3.8.1       
 [58] DEoptimR_1.0-8      MASS_7.3-51.1       zoo_1.8-5          
 [61] scales_1.0.0        doSNOW_1.0.16       parallel_3.5.3     
 [64] RColorBrewer_1.1-2  yaml_2.2.0          reticulate_1.11.1  
 [67] pbapply_1.4-0       gridExtra_2.3       rpart_4.1-13       
 [70] segmented_0.5-3.0   latticeExtra_0.6-28 stringi_1.4.3      
 [73] foreach_1.4.4       checkmate_1.9.1     caTools_1.17.1.2   
 [76] bibtex_0.4.2        Rdpack_0.10-1       SDMTools_1.1-221   
 [79] rlang_0.3.2         pkgconfig_2.0.2     dtw_1.20-1         
 [82] prabclus_2.2-7      bitops_1.0-6        evaluate_0.13      
 [85] lattice_0.20-38     ROCR_1.0-7          purrr_0.3.2        
 [88] labeling_0.3        htmlwidgets_1.3     bit_1.1-14         
 [91] tidyselect_0.2.5    plyr_1.8.4          magrittr_1.5       
 [94] R6_2.4.0            snow_0.4-3          gplots_3.0.1.1     
 [97] Hmisc_4.2-0         pillar_1.3.1        whisker_0.3-2      
[100] foreign_0.8-71      withr_2.1.2         fitdistrplus_1.0-14
[103] mixtools_1.1.0      survival_2.43-3     nnet_7.3-12        
[106] tsne_0.1-3          tibble_2.1.1        crayon_1.3.4       
[109] hdf5r_1.1.1         KernSmooth_2.23-15  rmarkdown_1.12     
[112] grid_3.5.3          data.table_1.12.0   git2r_0.25.2       
[115] metap_1.1           digest_0.6.18       diptest_0.75-7     
[118] R.utils_2.8.0       stats4_3.5.3        munsell_0.5.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
