<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>R Notebook</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">10x-adipocyte-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/PytrikFolkertsma/10x-adipocyte-analysis">
    <span class="fa fa-github"></span>
     
    Source
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>


<p><strong>Last updated:</strong> 2018-12-05</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20181026)</code> </summary></p>
<p>The command <code>set.seed(20181026)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/tree/9cba0fc31a590e8916063dc0e158689003c2e848" target="_blank">9cba0fc</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    output/markergenes/
    Ignored:    output/monocle/

Unstaged changes:
    Modified:   analysis/demuxlet.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/analysis/demuxlet.Rmd" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
<td style="text-align:left;">
demuxlet
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/PytrikFolkertsma/10x-adipocyte-analysis/9cba0fc31a590e8916063dc0e158689003c2e848/docs/demuxlet.html" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
<td style="text-align:left;">
demuxlet
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/PytrikFolkertsma/10x-adipocyte-analysis/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/demuxlet.html" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/18b27e4be2a50a5d8b797ecef721745176883ccc/analysis/demuxlet.Rmd" target="_blank">18b27e4</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
wflow_publish(“../analysis/demuxlet.Rmd”)
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Loading required package: cowplot</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre class="r"><code>data &lt;- readRDS(&#39;/projects/pytrik/sc_adipose/analyze_10x_fluidigm/10x-adipocyte-analysis/output/10x-180831&#39;)</code></pre>
<pre class="r"><code>getDemuxletForSample &lt;- function(i, outputdir){
  demuxlet &lt;- read.table(paste(&#39;/projects/pytrik/sc_adipose/analyze_10x_fluidigm/data/demuxlet/demuxlet_out/&#39;, outputdir, &#39;/180831_10x_s&#39;, i, &#39;.best&#39;, sep=&#39;&#39;), header=T)
  demuxlet$correct_barcode &lt;- paste(unlist(sapply(strsplit(as.character(demuxlet$BARCODE), &#39;-&#39;), &#39;[[&#39;, 1)), &#39;-&#39;, i, sep=&#39;&#39;) 
  rownames(demuxlet) &lt;- demuxlet$correct_barcode
  cells &lt;- rownames(data@meta.data)[data@meta.data$timepoint == paste(&#39;T&#39;, i, sep=&#39;&#39;)]
  demuxlet_filtered &lt;- demuxlet[demuxlet$correct_barcode %in% cells, ]
  
  num_sng_dbl &lt;- sapply(strsplit(as.character(demuxlet_filtered$BEST), &#39;-&#39;), &#39;[[&#39;, 1)
  print(paste(&#39;SAMPLE&#39;, i))
  print(table(num_sng_dbl))
  
  return(demuxlet_filtered)
}

getAllDemuxletResults &lt;- function(outdir, tutorialvcf=F){
  demuxlet_list &lt;- list()
  for (i in 1:5){
    demuxlet &lt;- getDemuxletForSample(i, outdir)
    demuxlet_list[[i]] &lt;- demuxlet
  }
  demuxlet_all &lt;- do.call(rbind, unname(demuxlet_list))
  demuxlet_all$label &lt;- as.character(demuxlet_all$BEST)
  demuxlet_all$label[startsWith(demuxlet_all$label, &#39;DBL&#39;)] &lt;- &#39;NA&#39;
  demuxlet_all$label[startsWith(demuxlet_all$label, &#39;AMB&#39;)] &lt;- &#39;NA&#39;
  
  hist(demuxlet_all$N.SNP)

  data &lt;- AddMetaData(data, demuxlet_all[&#39;label&#39;])

  print(paste(&#39;Average number of N.SNP:&#39;, mean(demuxlet_all$N.SNP)))
  print(paste(&#39;Number of cells: &#39;, dim(demuxlet_all)[1]))
  
  if (!tutorialvcf){
  plot_grid(
    DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;44B&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
    DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;BAT14&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
    DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;1AF&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
    DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;13a&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
    labels=c(&#39;44B&#39;, &#39;BAT14&#39;, &#39;1AF&#39;, &#39;13a&#39;)
  )
  } else{
    plot_grid(
      DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;293T&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
      DimPlot(data, cells.highlight = rownames(data@meta.data)[grep(&#39;jurkat&#39;, data@meta.data$label)], reduction.use=&#39;tsne&#39;, cols.use=&#39;gray&#39;, cols.highlight = &#39;blue&#39;),
      labels=c(&#39;239T&#39;, &#39;jurkat&#39;)
    )
  }
}</code></pre>
<div id="demuxlet-with-qced-vcf-file" class="section level1">
<h1>Demuxlet with QCed VCF file</h1>
<pre class="r"><code>getAllDemuxletResults(&#39;demuxlet_vcf_plink_bed-updated_sorted&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 1&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 3343   89 
[1] &quot;SAMPLE 2&quot;
num_sng_dbl
 AMB  DBL  SNG 
   9 5319  203 
[1] &quot;SAMPLE 3&quot;
num_sng_dbl
 AMB  DBL  SNG 
   9 5889  152 
[1] &quot;SAMPLE 4&quot;
num_sng_dbl
 AMB  DBL  SNG 
  25 4093  375 
[1] &quot;SAMPLE 5&quot;
num_sng_dbl
 AMB  DBL  SNG 
 156 6093 1116 </code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-4-1.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre><code>[1] &quot;Average number of N.SNP: 32.9931899374814&quot;
[1] &quot;Number of cells:  26872&quot;</code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-4-2.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-4-3.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-4.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-4-4.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-5.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-4-5.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="demuxlet-with-qced-vcf-file-imputed-snps-not-cleaned" class="section level1">
<h1>Demuxlet with QCed VCF file + imputed SNPS (NOT CLEANED)</h1>
<pre class="r"><code>getAllDemuxletResults(&#39;demuxlet_vcf-imputed-genotypes-NOTCLEANED&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 1&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 3423    9 
[1] &quot;SAMPLE 2&quot;
num_sng_dbl
 AMB  DBL  SNG 
   3 5514   14 
[1] &quot;SAMPLE 3&quot;
num_sng_dbl
 AMB  DBL  SNG 
   5 6034   11 
[1] &quot;SAMPLE 4&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 4476   16 
[1] &quot;SAMPLE 5&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 7278   86 </code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-5-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-5-1.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre><code>[1] &quot;Average number of N.SNP: 91.8610449538553&quot;
[1] &quot;Number of cells:  26872&quot;</code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-5-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-5-2.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-5-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-5-3.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-5-4.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-5-4.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-5-5.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/e50ea80bb2f39bbae91166c35248090f188c62d1/docs/figure/demuxlet.Rmd/unnamed-chunk-5-5.png" target="_blank">e50ea80</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="demuxlet-with-qced-vcf-file-imputed-snps-not-cleaned-alpha-0-alpha-0.5" class="section level1">
<h1>Demuxlet with QCed VCF file + imputed SNPS (NOT CLEANED), alpha 0 alpha 0.5</h1>
<p>Only did this for 3 samples. Same number of AMB, DBL and SNG as when running with default alpha settings.</p>
<pre class="r"><code>d1 &lt;- getDemuxletForSample(1, &#39;demuxlet_vcf-imputed-genotypes-alpha0.1-0.5-NOTCLEANED&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 1&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 3423    9 </code></pre>
<pre class="r"><code>d2 &lt;- getDemuxletForSample(2, &#39;demuxlet_vcf-imputed-genotypes-alpha0.1-0.5-NOTCLEANED&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 2&quot;
num_sng_dbl
 AMB  DBL  SNG 
   3 5514   14 </code></pre>
<pre class="r"><code>d3 &lt;- getDemuxletForSample(3, &#39;demuxlet_vcf-imputed-genotypes-alpha0.1-0.5-NOTCLEANED&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 3&quot;
num_sng_dbl
 AMB  DBL  SNG 
   5 6034   11 </code></pre>
</div>
<div id="demuxlet-with-qced-vcf-file-imputed-snps-cleaned-alpha-0-alpha-0.5" class="section level1">
<h1>Demuxlet with QCed VCF file + imputed SNPs, CLEANED, alpha 0 alpha 0.5</h1>
<p>Ran with cleaned up imputed VCF file (693,165 SNPs)</p>
<pre class="r"><code>getAllDemuxletResults(&#39;demuxlet_vcf-imputed-genotypes&#39;)</code></pre>
<pre><code>[1] &quot;SAMPLE 1&quot;
num_sng_dbl
 AMB  DBL  SNG 
  37 2963  431 
[1] &quot;SAMPLE 2&quot;
num_sng_dbl
 AMB  DBL  SNG 
  53 4688  788 
[1] &quot;SAMPLE 3&quot;
num_sng_dbl
 AMB  DBL  SNG 
  56 5361  630 
[1] &quot;SAMPLE 4&quot;
num_sng_dbl
 AMB  DBL  SNG 
   1 4454   37 
[1] &quot;SAMPLE 5&quot;
num_sng_dbl
 AMB  DBL  SNG 
  91 6000 1272 </code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-7-1.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre><code>[1] &quot;Average number of N.SNP: 33.3164693619239&quot;
[1] &quot;Number of cells:  26862&quot;</code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-7-2.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-7-3.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-4.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-7-4.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-5.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-7-5.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="demuxlet-with-tutorial-vcf-file" class="section level1">
<h1>Demuxlet with tutorial VCF file</h1>
<pre class="r"><code>getAllDemuxletResults(&#39;demuxlet_vcf_tutorial&#39;, tutorialvcf=T)</code></pre>
<pre><code>[1] &quot;SAMPLE 1&quot;
num_sng_dbl
 AMB  DBL  SNG 
 129  612 2692 
[1] &quot;SAMPLE 2&quot;
num_sng_dbl
 AMB  DBL  SNG 
 287  654 4590 
[1] &quot;SAMPLE 3&quot;
num_sng_dbl
 AMB  DBL  SNG 
 287  558 5205 
[1] &quot;SAMPLE 4&quot;
num_sng_dbl
 AMB  DBL  SNG 
 250  475 3768 
[1] &quot;SAMPLE 5&quot;
num_sng_dbl
 AMB  DBL  SNG 
 432 1884 5049 </code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-8-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-8-1.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre><code>[1] &quot;Average number of N.SNP: 185.059020541828&quot;
[1] &quot;Number of cells:  26872&quot;</code></pre>
<p><img src="figure/demuxlet.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-8-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-8-2.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-8-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/PytrikFolkertsma/10x-adipocyte-analysis/blob/9cba0fc31a590e8916063dc0e158689003c2e848/docs/figure/demuxlet.Rmd/unnamed-chunk-8-3.png" target="_blank">9cba0fc</a>
</td>
<td style="text-align:left;">
PytrikFolkertsma
</td>
<td style="text-align:left;">
2018-12-04
</td>
</tr>
</tbody>
</table>
<p></details> <img src="figure/demuxlet.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Storage

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Seurat_2.3.4  Matrix_1.2-14 cowplot_0.9.3 ggplot2_3.0.0

loaded via a namespace (and not attached):
  [1] Rtsne_0.13          colorspace_1.3-2    class_7.3-14       
  [4] modeltools_0.2-22   ggridges_0.5.0      mclust_5.4.1       
  [7] rprojroot_1.3-2     htmlTable_1.12      base64enc_0.1-3    
 [10] rstudioapi_0.7      proxy_0.4-22        flexmix_2.3-14     
 [13] bit64_0.9-7         mvtnorm_1.0-8       codetools_0.2-15   
 [16] splines_3.4.3       R.methodsS3_1.7.1   robustbase_0.93-2  
 [19] knitr_1.20          Formula_1.2-3       jsonlite_1.5       
 [22] workflowr_1.1.1     ica_1.0-2           cluster_2.0.7-1    
 [25] kernlab_0.9-27      png_0.1-7           R.oo_1.22.0        
 [28] compiler_3.4.3      httr_1.3.1          backports_1.1.2    
 [31] assertthat_0.2.0    lazyeval_0.2.1      lars_1.2           
 [34] acepack_1.4.1       htmltools_0.3.6     tools_3.4.3        
 [37] bindrcpp_0.2.2      igraph_1.2.2        gtable_0.2.0       
 [40] glue_1.3.0          RANN_2.6            reshape2_1.4.3     
 [43] dplyr_0.7.6         Rcpp_0.12.18        trimcluster_0.1-2.1
 [46] gdata_2.18.0        ape_5.1             nlme_3.1-137       
 [49] iterators_1.0.10    fpc_2.1-11.1        gbRd_0.4-11        
 [52] lmtest_0.9-36       stringr_1.3.1       irlba_2.3.2        
 [55] gtools_3.8.1        DEoptimR_1.0-8      MASS_7.3-50        
 [58] zoo_1.8-3           scales_1.0.0        doSNOW_1.0.16      
 [61] parallel_3.4.3      RColorBrewer_1.1-2  yaml_2.2.0         
 [64] reticulate_1.10     pbapply_1.3-4       gridExtra_2.3      
 [67] rpart_4.1-13        segmented_0.5-3.0   latticeExtra_0.6-28
 [70] stringi_1.2.4       foreach_1.4.4       checkmate_1.8.5    
 [73] caTools_1.17.1.1    bibtex_0.4.2        Rdpack_0.9-0       
 [76] SDMTools_1.1-221    rlang_0.2.2         pkgconfig_2.0.2    
 [79] dtw_1.20-1          prabclus_2.2-6      bitops_1.0-6       
 [82] evaluate_0.11       lattice_0.20-35     ROCR_1.0-7         
 [85] purrr_0.2.5         bindr_0.1.1         labeling_0.3       
 [88] htmlwidgets_1.2     bit_1.1-14          tidyselect_0.2.4   
 [91] plyr_1.8.4          magrittr_1.5        R6_2.2.2           
 [94] snow_0.4-2          gplots_3.0.1        Hmisc_4.1-1        
 [97] pillar_1.3.0        whisker_0.3-2       foreign_0.8-70     
[100] withr_2.1.2         fitdistrplus_1.0-9  mixtools_1.1.0     
[103] survival_2.42-6     nnet_7.3-12         tsne_0.1-3         
[106] tibble_1.4.2        crayon_1.3.4        hdf5r_1.0.0        
[109] KernSmooth_2.23-15  rmarkdown_1.10      grid_3.4.3         
[112] data.table_1.11.4   git2r_0.23.0        metap_1.0          
[115] digest_0.6.15       diptest_0.75-7      tidyr_0.8.1        
[118] R.utils_2.7.0       stats4_3.4.3        munsell_0.5.0      </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
