---
title: "R Notebook"
output: html_notebook
---


```{r}
library(Seurat)
library(monocle)
library(dplyr)
library(ggplot2)
library(ggrepel)
library(GOplot)
```

```{r}
seurobj <- readRDS('../output/seurat_objects/180831/10x-180831')
cds <- readRDS('../output/monocle/180831/10x-180831-monocle-monocle_genelist_T1T2T3_T4T5_res.1.5')
t1_aligned <- readRDS('../output/seurat_objects/180504/10x-180504-ccregout-aligned')
source('../code/colors.R')
```

#Figure 1

```{r echo=T, results='hide', include=F}
p1 <- TSNEPlot(seurobj, group.by='depot', colors.use=colors.depots, pt.size=0.1, no.axes=T)

p2 <- TSNEPlot(seurobj, group.by='timepoint', colors.use=colors.timepoints, pt.size=0.5, no.axes=F, do.return=T) + theme(axis.text = element_blank(), axis.ticks = element_blank(), axis.title = element_blank(), plot.margin=grid::unit(c(0,0,0,0), "mm"))

t1_aligned <- SetAllIdent(t1_aligned, id='depot')
t1_aligned@ident <- factor(t1_aligned@ident, levels=c('Supra', 'Subq', 'Peri', 'Visce'))

p3 <- TSNEPlot(t1_aligned, group.by='depot', colors.use=colors.depots, do.return=T, pt.size=0.5, no.axes=T)

p3$data <- p3$data %>% mutate(ident=factor(ident, levels=c('Visce', 'Supra', 'Subq', 'Peri'))) %>% arrange

fig1 <- plot_grid(p1, p2, p3, ncol=2)
```

```{r, fig.height = 10, fig.width = 12, fig.align = "center"}
fig1
```

```{r}
save_plot("../figures/figures_paper/main_figures/Figure1_overview/10x-180831_tsne_depots.pdf", p1, base_width=6, base_height=4.5)
save_plot("../figures/figures_paper/main_figures/Figure1_overview/10x-180831_tsne_timepoints.pdf", p2, base_width=4, base_height=3)
save_plot("../figures/figures_paper/main_figures/Figure1_overview/10x-180831_tsne_t1_aligned.pdf", p3, base_width=6, base_height=4.5)
```


#Figure 2


```{r echo=T, results='hide', include=F}
#10x-180831_tsne_depots
#10x-180831_tsne_timepoints
#10x-180831_trajectory_depots
#10x-180831_trajectory_timepoints
#10x-180831_tsne_state

traj_T <- plot_cell_trajectory(cds, color_by = "timepoint") + geom_point(color='white', size=5) + geom_point(aes(colour=timepoint), alpha=0.1) + scale_color_manual(values=colors.timepoints) + theme(legend.position='bottom')

tsne_branches <- TSNEPlot(seurobj, group.by='State.labels', colors.use=colors.states.labels, no.axes=T, pt.size=pt.size.tsne) + theme(legend.justification='top')

traj_pt <- plot_cell_trajectory(cds, color_by='Pseudotime_stretched', show_branch_points = F, show_backbone=F) + scale_colour_gradient(low='#90d8f2', high='#006488', name='Pseudotime\n(stretched)', breaks=c(0, 50, 100)) + theme_void() + theme(legend.position = 'right', legend.text=element_text(size=16), legend.title=element_text(size=16), legend.key.size=unit(0.6, 'cm'))

fp_pt <- FeaturePlot(seurobj, features.plot='Pseudotime_stretched', cols.use=c('#90d8f2', '#00719a'), do.return=T, no.axes=T, pt.size=1, no.legend=F)$Pseudotime_stretched + scale_colour_gradient(low='#90d8f2', high='#006488', name='Pseudotime\n(stretched)') + theme(plot.title=element_blank(), legend.title=element_text(size=20), legend.text=element_text(size=20), legend.key.height=unit(1.3, 'cm'))

tsne_depot <- TSNEPlot(seurobj, group.by='depot', colors.use=colors.depots, pt.size=pt.size.tsne, no.axes=F) + theme(legend.text=element_text(size=13), legend.key.height=unit(0.7, 'cm'), axis.text = element_blank(), axis.ticks = element_blank(), axis.title = element_blank(), plot.margin=grid::unit(c(0,0,0,0), "mm"))

fig2 <- plot_grid(
  traj_T, tsne_branches, traj_pt, fp_pt, tsne_depot, ncol=2
)
```

```{r, fig.height = 15, fig.width = 12, fig.align = "center"}
fig2
```


```{r echo=T, results='hide', include=F}
#traj_T, tsne_branches, traj_pt, fp_pt, tsne_depot, ncol=2
save_plot("../figures/figures_paper/main_figures/Figure2_monocle/10x-180831_trajectory_timepoints.pdf", traj_T, base_width=6, base_height=4.5)

save_plot("../figures/figures_paper/main_figures/Figure2_monocle/10x-180831_tsne_branch.pdf", tsne_branches, base_width=5, base_height=4.5)

save_plot("../figures/figures_paper/main_figures/Figure2_monocle/10x-180831_trajectory_pseudotime.pdf", traj_pt, base_width=5, base_height=2)

save_plot("../figures/figures_paper/main_figures/Figure2_monocle/10x-180831_tsne_pseudotime.pdf", fp_pt, base_width=5.2, base_height=3.5)

save_plot("../figures/figures_paper/main_figures/Figure2_monocle/10x-180831_tsne_depot.pdf", tsne_depot, base_width=3.3, base_height=2.25)
```


#Figure 3

##Heatmap

Heatmap

```{r, fig.height = 7, fig.width = 6, fig.align = "center"}
load('../output/monocle/180831/heatmaps')
heatmap <- heatmaps$heatmap_logFC0.3_ncluster6_filtered_genes
gridExtra::grid.arrange(heatmap$ph_res$gtable)
```

```{r}
save_plot("../figures/figures_paper/main_figures/figure3/BEAM_heatmap_logFC0.3_ncluster6_filtered_genes.pdf", heatmap$ph_res$gtable, base_width=7, base_height=8)
```

##Pseudotime plots

Pseudotime plots

```{r}
#new_cds <- buildBranchCellDataSet(cds, progenitor_method='duplicate', branch_states=c(2,3), branch_point=1, branch_labels=c('Metabolic', 'ECM'), stretch=T)
#saveRDS(new_cds, file='../output/monocle/180831/10x-180831-monocle-monocle_genelist_T1T2T3_T4T5_res.1.5_duplicated-progenitor_stretched-pseudotime')

new_cds <- readRDS('../output/monocle/180831/10x-180831-monocle-monocle_genelist_T1T2T3_T4T5_res.1.5_duplicated-progenitor_stretched-pseudotime')
#new cds contains duplicated progenitor cells (one assigned to each branch) and stretched Pseudotime

df_pdata <- pData(new_cds)[c('original_cell_id', 'Branch', 'State.labels', 'Pseudotime')]

#get expression values for every cell ID from the new cds
genes <- as.matrix(t(seurobj@data[c('ADIPOQ', 'MGP', 'UCP2', 'DCN', 'APOD'), df_pdata$original_cell_id]))

#cbind the expression values with the relevant metadata
df <- cbind(df_pdata, as.data.frame(genes))
```


```{r, fig.height = 8, fig.width = 10, fig.align = "center"}
plot_pseudotime <- function(gene, legend){
  p <- ggplot(df, aes_string(x='Pseudotime', y=gene)) +
    geom_point(shape = 21, colour = "black", size = 1, stroke = 0, alpha=0.2, aes(fill=State.labels), show.legend=F)
  
  if (legend){
    p <- p + geom_smooth(se = FALSE, aes(color=Branch), span=0.9, method='loess', size=2)
  } else {
    p <- p + geom_smooth(se = FALSE, aes(color=Branch), span=0.9, method='loess', size=2, show.legend=F)
  }
  p <- p + 
    scale_color_manual(values=colors.pseudotime.lines) +
    scale_fill_manual(values=colors.pseudotime.points) +
    scale_y_continuous(minor_breaks=1) +
    xlab('Pseudotime (stretched)') +
    ggtitle(gene) +
    ylab('Expression') +
    theme(plot.title = element_text(size=30), legend.text=element_text(size=30), axis.text=element_text(size=15), axis.title=element_text(size=20), legend.title=element_text(size=30), legend.key.size=unit(1.5, 'cm'))
  return(p)
}

adipoq_pt <- plot_pseudotime('ADIPOQ', legend=F)
ucp2_pt <- plot_pseudotime('UCP2', legend=T)
dcn_pt <- plot_pseudotime('DCN', legend=F)
apod_pt <- plot_pseudotime('APOD', legend=T)

metabolic_pt <- plot_grid(adipoq_pt, ucp2_pt, ncol=2, rel_widths=c(0.415, 0.585))
ecm_pt <- plot_grid(dcn_pt, apod_pt, ncol=2, rel_widths=c(0.415, 0.585))

plot_grid(metabolic_pt, ecm_pt, ncol=1)
```

```{r}
width=12
height=5

save_plot("../figures/figures_paper/main_figures/Figure3_beam/pseudotime_metabolic.pdf", metabolic_pt, base_width=width, base_height=height)
save_plot("../figures/figures_paper/main_figures/Figure3_beam/pseudotime_ecm.pdf", ecm_pt, base_width=width, base_height=height)


#save_plot("../figures/figures_paper/main_figures/figure3/pseudotime_ADIPOQ.pdf", adipoq_pt, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/pseudotime_UCP2.pdf", ucp2_pt, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/pseudotime_DCN.pdf", dcn_pt, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/pseudotime_APOD.pdf", apod_pt, base_width=width, base_height=height)
```

##Featureplots

```{r echo=T, results='hide'}
genes <- c('ADIPOQ', 'UCP2', 'DCN', 'APOD')
featureplots_noleg <- FeaturePlot(seurobj, features.plot=genes, cols.use=c('gray', 'blue'), no.legend=T, do.return=T, no.axes=T, pt.size=1, max.cutoff=c(5)) 
featureplots_leg <- FeaturePlot(seurobj, features.plot=genes, cols.use=c('gray', 'blue'), no.legend=F, do.return=T, no.axes=T, pt.size=1, max.cutoff=c(5)) 

adipoq_fp <- featureplots_noleg$ADIPOQ + theme(plot.title=element_blank())

ucp2_fp <- featureplots_leg$UCP2 + scale_color_gradient(name='Expression', low='gray', high='blue', guide='colorbar', limits=c(0,5)) + theme(plot.title=element_blank(), legend.title=element_text(size=20), legend.text=element_text(size=20), legend.key.height = unit(1.3, 'cm'))

dcn_fp <- featureplots_noleg$DCN + theme(plot.title=element_blank())

apod_fp <- featureplots_leg$APOD + scale_color_gradient(name='Expression', low='gray', high='blue', guide='colorbar') + theme(plot.title=element_blank(), legend.title=element_text(size=20), legend.text=element_text(size=20), legend.key.height = unit(1.3, 'cm'))
```

```{r, fig.height = 10, fig.width = 12, fig.align = "center"}
metabolic_fp <- plot_grid(adipoq_fp, ucp2_fp, ncol=2, rel_widths=c(0.41, 0.59))
ecm_fp <- plot_grid(dcn_fp, apod_fp, ncol=2, rel_widths=c(0.41, 0.59))

plot_grid(metabolic_fp, ecm_fp, ncol=1)
```


```{r}

save_plot("../figures/figures_paper/main_figures/Figure3_beam/featureplots_metabolic.pdf", metabolic_fp, base_width=width, base_height=height)
save_plot("../figures/figures_paper/main_figures/Figure3_beam/featureplots_ecm.pdf", ecm_fp, base_width=width, base_height=height)

#save_plot("../figures/figures_paper/main_figures/figure3/featureplot_ADIPOQ.pdf", adipoq_fp, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/featureplot_UCP2.pdf", ucp2_fp, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/featureplot_DCN.pdf", dcn_fp, base_width=width, base_height=height)
#save_plot("../figures/figures_paper/main_figures/figure3/featureplot_APOD.pdf", apod_fp, base_width=width, base_height=height)
```

##GO bubble plot

```{r}
name <- 'heatmap_logFC0.3_ncluster6'
folder <- paste('../tables/tables_paper/all_tables/BEAM/', name, sep='')
#clusters <- c(1,2,3,4,6)
clusters <- c(2,6,3)
gsea <- list()
genelist <- list()

for (i in clusters){
  gsea_cluster <- read.table(paste(folder, '/REVIGO_cleaned/', 'REVIGO_cluster', i, '_cleaned.tsv', sep=''), header=T)
  gsea_cluster['cluster'] <- i
  genes_cluster <- read.table(paste(folder, '/genelist_cluster', i, '.tsv', sep=''), header=T)
  gsea[[i]] <- gsea_cluster
  genelist[[i]] <- genes_cluster
}

gsea <- do.call("rbind", gsea)
genelist <- do.call("rbind", genelist)

create_terms <- function(gsea){
  return(data.frame(category=gsea$gsea.domain, ID=gsea$revigo.term_ID, term=gsea$revigo.description, adj_pval=gsea$gsea.p.value, genes=gsea$gsea.intersection, cluster=gsea$cluster))
}

create_genes <- function(genelist){
  return(data.frame(ID=genelist$gene_short_name, logFC=genelist$avgLogFC_State2_State3))
}

terms <- create_terms(gsea)
genes <- create_genes(genelist)
circ <- circle_dat(terms, genes)

```

```{r, fig.height = 7, fig.width = 20, fig.align = "center"}
GOBubble_modified <- function(data, display, title, colour, labels, ID = T, table.legend = T, table.col = T, bg.col = F, zscore_threshold=0, ggrepel_linecol='black', text_size_axes=20, text_size_axes_label=20, text_size=10, text_size_legend=7, text_size_threshold=10, legend_point_size=20){
  zscore <- adj_pval <- category <- count <- id <- term <- NULL
  if (missing(display)) display <- 'single'
  if (missing(title)) title <- ''
  if (missing(colour)) cols <- c("chartreuse4", "brown2", "cornflowerblue") else cols <- colour
  if (missing(labels)) labels <- 5
  if (bg.col == T & display == 'single') cat("Parameter bg.col will be ignored. To use the parameter change display to 'multiple'")
  
  colnames(data) <- tolower(colnames(data))
  tmp_lab <- data.frame(cats = c('Biological Process', 'Cellular Component', 'Molecular Function'), 
                        cat.abb = c('BP', 'CC', 'MF'),
                        locy = c(match('BP', data$category), match('CC', data$category), match('MF', data$category)),
                        cols = cols,
                        stringsAsFactors = F)
  tmp_lab <- tmp_lab[order(tmp_lab$locy),]
  if(!'count'%in%colnames(data)){
    rang <- c(5, 5)
    data$count <- rep(1, dim(data)[1])
  }else {rang <- c(1, 30)}
  data$adj_pval <- -log(data$adj_pval, 10)
  sub <- data[!duplicated(data$term), ]
  g <- ggplot(sub, aes(zscore, adj_pval, fill = category, size = count))+
    labs(title = title, x = 'z-score', y = '-log (adj p-value)')+
    geom_point(shape = 21, col = 'black', alpha = 1 / 2)+ #add all bubbles
    geom_hline(yintercept = 1.3, col = 'orange')+ #add threshold for p-value=0.05
    scale_size(range = rang, guide = 'none') +
    scale_y_continuous(limits=c(0,45))
  
  sub2 <- subset(sub, subset=sub$adj_pval >= labels)
  sub2 <- subset(sub2, subset=abs(sub2$zscore) >= zscore_threshold)

    g <- g + scale_fill_manual('', values = tmp_lab$cols, labels = tmp_lab$cats)+
      theme(legend.position = 'right', legend.key.size = unit(1, 'cm'), legend.title=element_text(face='bold')) +
      guides(fill = guide_legend(override.aes = list(size=legend_point_size)))
      annotate ("text", x = 0.6, y = 2.2, label = "Threshold", colour = "orange", size = text_size_threshold)
    g <- g + geom_text_repel(data = sub2, aes(x = zscore, y = adj_pval, label = term), size = text_size, segment.color = ggrepel_linecol)
    g <-  g + 
        theme(axis.text = element_text(size = text_size_axes), axis.line = element_line(colour = 'grey80'), axis.ticks = element_line(colour = 'grey80'), axis.title = element_text(size = text_size_axes_label, face = 'bold'), panel.background = element_blank(), panel.grid.minor = element_blank(), panel.grid.major = element_blank(), plot.background = element_blank(), legend.text=element_text(size=text_size_legend), legend.title=element_text(size=text_size_legend))
 return(g)
}

p <- GOBubble_modified(circ, labels=2, ID=F, table.legend = F, zscore_threshold=4.5, ggrepel_linecol='transparent', text_size_axes=15, text_size_axes_label=15, text_size=4.5, text_size_legend=15, text_size_threshold=5, legend_point_size=15)
p
```

```{r}
save_plot('../figures/figures_paper/main_figures/Figure3_beam/180831_beamGO_cluster-2-6-3.pd_labels3_zscore4.5.pdf', p, base_width=20, base_height=7)
#save_plot('../figures/figures_paper/main_figures/Figure3_beam/180831_beamGO_all-terms.pd_labels3_zscore4.5.pdf', p, base_width=20, base_height=7)
```

#Percent.mito

```{r echo=T, results='hide'}
percent_mito_fp <- FeaturePlot(seurobj, features.plot=c('percent.mito.100'), cols.use=c('gray', 'blue'), no.legend=F, do.return=T, no.axes=T, max.cutoff = c(10))$percent.mito + ggtitle('') + scale_color_gradient(name='% mitochondrial\ngene expression', low='gray', high='blue', guide='colorbar') + theme(legend.key.height = unit(1.6, 'cm'))
```

```{r, fig.height = 4, fig.width = 5.5, fig.align = "center"}
percent_mito_fp
```

```{r}
save_plot("../figures/figures_paper/main_figures/Figure3_beam/featureplot_PERCENT_MITO.pdf", percent_mito_fp, base_width=width, base_height=height)
```

#BATLAS

```{r}
batlas_results <- read.table('../tables/tables_paper/supplementary_tables/BATLAS/BATLAS_branch_high_res.txt', sep='\t', header=T)
batlas_results['pseudotime_decile'] <- seurobj@meta.data$pseudotime_decile[match(batlas_results$sample, seurobj@meta.data$branch_high_res)]
batlas_results['branch'] <- seurobj@meta.data$branch[match(batlas_results$sample, seurobj@meta.data$branch_high_res)]
batlas_results <- batlas_results %>% mutate(block_branch = if_else(branch == 'Progenitor', 'Beginning', 'End'))
batlas_results['brown_centered'] <- batlas_results$brown - 0.5
batlas_results['ratio'] <- batlas_results$brown/batlas_results$white
```

```{r, fig.height = 6, fig.width = 8, fig.align = "center"}
p <- ggplot(batlas_results, aes(x=as.factor(pseudotime_decile), y=brown_centered, fill=branch)) +
  geom_col(position = position_dodge2(width=0.9, preserve='single')) +
  labs(x="Pseudotime (stretched)", fill="Branch", y=NULL) +
  scale_fill_manual(values=colormap.branches, labels=c('L', 'U', 'P')) +
  facet_grid(cols=vars(block_branch), switch='x') + 
  scale_y_continuous(limits=c(-0.5, 0.5), breaks=seq(-0.5, 0.5, by=0.5), labels=c('100% white', '50/50', '100% brown')) +
  theme(strip.background=element_blank(), strip.placement='outside', strip.text = element_blank(), axis.text.x = element_blank(), axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))
p
```

```{r}
save_plot('../figures/figures_paper/main_figures/Figure3_beam/BATLAS_results.pdf', p, base_height=5, base_width=8)
```


##SOM Figures

Clusters

TF analysis

```{r, fig.height = 10, fig.width = 12, fig.align = "center"}
fig <- plot_grid(
  TSNEPlot(seurobj, group.by='res.0.5', pt.size=pt.size.tsne, do.label=T),
  TSNEPlot(seurobj, group.by='res.0.7', pt.size=pt.size.tsne, do.label=T),
  TSNEPlot(seurobj, group.by='res.1', pt.size=pt.size.tsne, do.label=T),
  TSNEPlot(seurobj, group.by='res.1.5', pt.size=pt.size.tsne, do.label=T),
  ncol=2, labels='auto'
)
```


Metabolic and ECM genes

```{r}

```


```{r}
FeaturePlot(seurobj, features.plot='Pseudotime')
```

